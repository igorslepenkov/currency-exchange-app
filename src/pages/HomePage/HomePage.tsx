import {
  Box,
  Button,
  Typography,
  Paper,
  Theme,
  ButtonGroup,
} from "@mui/material";
import { SxProps } from "@mui/system";
import { useNavigate } from "react-router-dom";
import { ROUTE, RouteKey } from "../../router";

export const HomePage = () => {
  const pageStyles: SxProps = {
    position: "relative",
    flexGrow: 1,
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "center",
  };

  const greetingTextStyles: SxProps = {
    "@keyframes appear": {
      "0%": {
        opacity: 0,
      },
      "20%": {
        opacity: 0.2,
      },
      "40%": {
        opacity: 0.4,
      },
      "60%": {
        opacity: 0.6,
      },
      "80%": {
        opacity: 0.8,
      },
      "100%": {
        opacity: 1,
      },
    },

    "@keyframes dissapear": {
      "0%": {
        opacity: 1,
      },
      "20%": {
        opacity: 0.8,
      },
      "40%": {
        opacity: 0.6,
      },
      "60%": {
        opacity: 0.4,
      },
      "80%": {
        opacity: 0.2,
      },
      "100%": {
        opacity: 0,
        display: "none",
      },
    },
    opacity: 0,
    animationName: "appear, dissapear",
    animationDuration: "3s, 3s",
    animationDelay: "0s, 3s",
    animationFillMode: "forwards, forwards",
    animationTimingFunction: "ease-in-out, ease-in-out",
    textAlign: "center",
  };

  const panelStyles: SxProps<Theme> = (theme) => ({
    "@keyframes appear": {
      "0%": {
        opacity: 0,
      },
      "20%": {
        opacity: 0.2,
      },
      "40%": {
        opacity: 0.4,
      },
      "60%": {
        opacity: 0.6,
      },
      "80%": {
        opacity: 0.8,
      },
      "100%": {
        opacity: 1,
      },
    },
    position: "absolute",
    top: "50%",
    left: "50%",
    transform: "translate(-50%, -50%)",
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "center",
    height: "300px",
    width: "300px",
    padding: "10px",
    borderRadius: "20px",
    opacity: 0,
    animationName: "appear",
    animationDelay: "6.5s",
    animationDuration: "2s",
    animationFillMode: "forwards",
    animationTimingFunction: "ease-in-out",
    [theme.breakpoints.up("sm")]: {
      width: "400px",
    },
    [theme.breakpoints.up("md")]: {
      width: "600px",
    },
  });

  const navigate = useNavigate();

  return (
    <Box sx={pageStyles} component="main">
      <Typography variant="h2" color="secondary" sx={greetingTextStyles}>
        Welcome to Exchage App!
      </Typography>

      <Typography
        variant="h4"
        sx={{ ...greetingTextStyles, animationDelay: "1s, 4s" }}
      >
        We are glad to see you!
      </Typography>
      <Paper sx={panelStyles}>
        <Typography variant="h4" mb="15px">
          Choose page:
        </Typography>
        <ButtonGroup variant="contained" size="large" fullWidth>
          <Button
            color="secondary"
            sx={{ flexGrow: 1 }}
            type="button"
            onClick={() => navigate(ROUTE[RouteKey.Converter])}
          >
            Converter
          </Button>
          <Button
            color="secondary"
            sx={{ flexGrow: 1 }}
            type="button"
            onClick={() => navigate(ROUTE[RouteKey.Currencies])}
          >
            List of Currencies
          </Button>
        </ButtonGroup>
      </Paper>
    </Box>
  );
};
