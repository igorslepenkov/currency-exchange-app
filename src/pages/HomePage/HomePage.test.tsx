import { render } from "@testing-library/react";
import { HomePage } from "./HomePage";

describe(HomePage, () => {
  it("Renders HomePage correctly", () => {
    const tree = render(<HomePage />);

    expect(tree).toMatchSnapshot();
  });
});
