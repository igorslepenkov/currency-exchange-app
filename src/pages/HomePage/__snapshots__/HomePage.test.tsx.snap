// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Renders HomePage correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="MuiBox-root css-1exp1tx"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 css-1jmbp4u-MuiTypography-root"
        >
          Welcome to Exchage App!
        </h2>
        <h4
          class="MuiTypography-root MuiTypography-h4 css-1xwpc42-MuiTypography-root"
        >
          We are glad to see you!
        </h4>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1wfvmer-MuiPaper-root"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 css-1h02dxw-MuiTypography-root"
          >
            Choose page:
          </h4>
          <div
            class="MuiButtonGroup-root MuiButtonGroup-contained MuiButtonGroup-fullWidth css-1ys9lua-MuiButtonGroup-root"
            role="group"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-fullWidth MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-fullWidth MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary css-xut1cy-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Converter
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-fullWidth MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-fullWidth MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary css-xut1cy-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              List of Currencies
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="MuiBox-root css-1exp1tx"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 css-1jmbp4u-MuiTypography-root"
      >
        Welcome to Exchage App!
      </h2>
      <h4
        class="MuiTypography-root MuiTypography-h4 css-1xwpc42-MuiTypography-root"
      >
        We are glad to see you!
      </h4>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1wfvmer-MuiPaper-root"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-1h02dxw-MuiTypography-root"
        >
          Choose page:
        </h4>
        <div
          class="MuiButtonGroup-root MuiButtonGroup-contained MuiButtonGroup-fullWidth css-1ys9lua-MuiButtonGroup-root"
          role="group"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-fullWidth MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-fullWidth MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary css-xut1cy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Converter
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-fullWidth MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-fullWidth MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary css-xut1cy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            List of Currencies
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
